src/guiguts/mainwindow.py

```diff
@@ -363,7 +363,9 @@ def do_save(self, fname: str) -> None:
             fname: Name of file to save text to.
         """
         with open(fname, "w", encoding="utf-8") as fh:
-            fh.write(self.get(1.0, tk.END))
+            fh.write(
+                self.get(1.0, f"{tk.END}-1c")
```

Comment:
Feels a bit weird to have this "get all the text" not match the `get_md5_checksum()`'s version. May not be worth it, but consider having a function in this class that can be used to fetch the window's text contents so we don't need to implement that in different places.

It might be clearer to `.rstrip()` the text of trailing newlines.